sphinx/util/nodes.py

```diff
@@ -41,13 +41,19 @@ def apply_source_workaround(node):
     # * rawsource of term node will have: ``term text : classifier1 : classifier2``
     # * rawsource of classifier node will be None
     if isinstance(node, nodes.classifier) and not node.rawsource:
+        logger.debug('[i18n] PATCH: %r to have source, line and rawsource: %s',
+                     logging.get_full_module_name(node), logging.repr_domxml(node))
         definition_list_item = node.parent
         node.source = definition_list_item.source
         node.line = definition_list_item.line - 1
         node.rawsource = node.astext()  # set 'classifier1' (or 'classifier2')
     if isinstance(node, nodes.image) and node.source is None:
+        logger.debug('[i18n] PATCH: %r to have source, line: %s',
+                     logging.get_full_module_name(node), logging.repr_domxml(node))
```

Comment:
@shimizukawa 
I don't know why I didn't think to ask this sooner, why `get_full_module_name()` and `repr_domxml()` are placed on  logging module? I think both are utilities for nodes. How about moving them to `sphinx.util.nodes`?

Reply:
OK, I'll move it soon.